packages/in_app_purchase/lib/src/in_app_purchase_connection/product_details.dart

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// The class represents the information of a product.
+///
+/// A list of [ProductDetails] can be obtained from the [QueryProductDetailsResponse].
+class ProductDetails {
+  ProductDetails({
+    @required this.id,
+    @required this.title,
+    @required this.description,
+    @required this.price,
+  });
+
+  /// The identifier of the product, specified in App Store Connect or Sku in Google Play console.
+  final String id;
+
+  /// The title of the product, specified in the App Store Connect or Sku in Google Play console based on the platform.
+  final String title;
+
+  /// The description of the product, specified in the App Store Connect or Sku in Google Play console based on the platform.
+  final String description;
+
+  /// The price of the product, specified in the App Store Connect or Sku in Google Play console based on the platform.
+  /// Formatted with currency symbol ("$0.99").
+  final String price;
+}
+
+/// The response returned by [InAppPurchaseConnection.queryProductDetails]
+class QueryProductDetailsResponse {
```

Comment:
nit: I think it would be a little clearer and more concise to name this `ProductDetailsResponse`. The "Query" part is already implied by "Response" (it's a response to a query).